src/pages/LoginView.jsx

```diff
@@ -28,74 +24,89 @@ function LoginView() {
-    }
+  function emailValidationCheck(email) {
+    setEmail(email);
+    handleEmailValidationResponse(email);
+  }
+
+  function passwordValidationCheck(password) {
+    setPassword(password);
+    handlePasswordValidationResponse(password);
+  }
 
-    return (emailResponse === SUCCESS) && (passwordResponse === SUCCESS);
+  /* Takes in validation response of email and sets based on success or not
+   *
+   * @param response
+   */
+  function handleEmailValidationResponse(input) {
+    const message = input === '' ? ERROR_EMPTY_EMAIL : null;
+    setEmailErrorMessage(message);
   }
-  
+
+  /* Takes in validation response of password and sets based on success or not
+   *
+   * @param response
+   */
+  function handlePasswordValidationResponse(input) {
+    const message = input === '' ? ERROR_EMPTY_PASSWORD : null;
+    setPasswordErrorMessage(message);
+  }
+
   /************************************
    * Render
    ************************************/
 
   return (
     <div className='authentication-view-body'>
-        <div className='authentication-input-container'> 
-            <img src={homeLogo} alt="oneleif logo" />
-            <div className="form-container">
-              <Input 
-                className='auth'
-                label='Email'
-                onValueChange={(email) => setEmail(email)} 
-                errorMessage={emailErrorMessage}/>
-              <Input 
-                className='auth'
-                label='Password'
-                type='password'
-                onValueChange={(password) => setPassword(password)} 
-                errorMessage={passwordErrorMessage}/>
-              <div className='authentication-actions-module'>
-                <span>Forgot your password?</span>
-                <button onClick={() => loginClicked()}>Log in</button>
-              </div>
-            </div>
-            {errorMessage && <p className='error-message'>{errorMessage}</p>}
+      <div className='authentication-input-container'>
+        <img src={homeLogo} alt='oneleif logo' />
+        <div className='form-container'>
+          <Input
+            className='auth'
+            label='Email'
+            onValueChange={email => emailValidationCheck(email)}
+            errorMessage={emailErrorMessage}
+          />
+          <Input
+            className='auth'
+            label='Password'
+            type='password'
+            onValueChange={password => passwordValidationCheck(password)}
+            errorMessage={passwordErrorMessage}
+          />
+          <div className='authentication-actions-module'>
+            <span>Forgot your password?</span>
+            <button onClick={() => loginClicked()}>Log in</button>
+          </div>
         </div>
-        {isLoggedIn && <Redirect to="/" />}
+        {errorMessage && <p className='error-message'>{errorMessage}</p>}
+      </div>
+      {isLoggedIn && <Redirect to={props.location.state?.referer || '/'} />}
```

Comment:
```props.location.state?.referer``` The optional chaining in this is showing errors in my IDE but not in the terminal. Is there any way to get rid of this?

Reply:
Support for optional chaining in IDEs is still at a somewhat initial stage, so getting rid of the error would be dependent on the specific IDE. 

VS Code version 1.41 supports optional chaining out of the box.